src/ProtectedProxies/ProtectedBeaconProxy.sol

```diff
@@ -22,4 +24,12 @@ contract ProtectedBeaconProxy is SphereXProtectedProxy, BeaconProxy {
     function _delegate(address implementation) internal virtual override(Proxy, SphereXProtectedProxy) {
         SphereXProtectedProxy._delegate(implementation);
     }
+
+    function _sphereXEngine() internal view override returns (ISphereXEngine) {
```

Comment:
This will create 3 delegate calls per 1 call to the proxy. Don't you think this is too much?
I think it's preferable to consolidate them to 1 call